examples/platforms/posix/flash.c

```diff
@@ -0,0 +1,149 @@
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <string.h>
+
+#include <openthread-config.h>
+#include <platform/flash.h>
+
+#include <common/code_utils.hpp>
+#include "platform-posix.h"
+
+static int sFlashFd;
+uint32_t sEraseAddress;
+
+enum
+{
+    FLASH_SIZE = 0x40000,
+    FLASH_PAGE_SIZE = 0x800,
+    FLASH_PAGE_NUM = 128,
+};
+
+ThreadError otPlatFlashInit(void)
+{
+    ThreadError error = kThreadError_None;
+    char fileName[16];
+    struct stat st;
+    bool create = false;
+
+    memset(&st, 0, sizeof(st));
+
+    if (stat("tmp", &st) == -1)
+    {
+        mkdir("tmp", 0777);
+    }
+
+    snprintf(fileName, sizeof(fileName), "tmp/%d.flash", NODE_ID);
+
+    if (access(fileName, 0))
+    {
+        create = true;
+    }
+
+    sFlashFd = open(fileName, O_RDWR | O_CREAT, 0666);
+    lseek(sFlashFd, 0, SEEK_SET);
+
+    VerifyOrExit(sFlashFd >= 0, error = kThreadError_Failed);
+
+    if (create)
+    {
+        for (uint16_t index = 0; index < FLASH_PAGE_NUM; index++)
+        {
+            SuccessOrExit(error = otPlatFlashErasePage(index * FLASH_PAGE_SIZE));
+        }
+    }
+
+exit:
+    return error;
+}
+
+uint32_t otPlatFlashGetSize(void)
+{
+    return FLASH_SIZE;
+}
+
+ThreadError otPlatFlashErasePage(uint32_t aAddress)
+{
+    ThreadError error = kThreadError_None;
+    uint8_t buf = 0xff;
+    uint32_t address;
+
+    VerifyOrExit(sFlashFd >= 0, error = kThreadError_Failed);
+    VerifyOrExit(aAddress < FLASH_SIZE, error = kThreadError_InvalidArgs);
+
+    address = aAddress & (~(uint32_t)(FLASH_PAGE_SIZE - 1));
+
+    for (uint16_t offset = 0; offset < FLASH_PAGE_SIZE; offset++)
+    {
+        VerifyOrExit(pwrite(sFlashFd, &buf, 1, address + offset) == 1, error = kThreadError_Failed);
+    }
+
+exit:
+    return error;
+}
+
+ThreadError otPlatFlashStatusWait(uint32_t aTimeout)
+{
+    (void)aTimeout;
+    return kThreadError_None;
+}
+
+uint32_t otPlatFlashWrite(uint32_t aAddress, uint8_t *aData, uint32_t aSize)
+{
+    uint32_t ret = 0;
+    uint32_t index = 0;
+    uint8_t byte;
+
+    VerifyOrExit(sFlashFd >= 0 && aAddress < FLASH_SIZE, ;);
+
+    for (index = 0; index < aSize; index++)
+    {
+        VerifyOrExit((ret = otPlatFlashRead(aAddress + index, &byte, 1)) == 1, ;);
+        byte &= aData[index];
```

Comment:
I'd add a comment here explaining why we use `&=` instead of `=`. Something like:

    // Use bitwise AND to emulate behavior of flash memory